hydrus/data/crud.py

```diff
@@ -186,7 +189,39 @@ def insert(object_: Dict[str, Any], session: scoped_session,
                 # Adds new Property
                 session.close()
                 raise PropertyNotFound(type_=prop_name)
-
+            # For insertion in III through link
+            if prop_name in link_props:
+                regex = r'[a-z0-9]{8}-([a-z0-9]{4}-){3}[a-z0-9]{12}'
+                matchObj = re.match(regex, link_props[prop_name])
+                # Link is to an instance of a collection class
+                if matchObj:
+                    try:
+                        nested_instance = session.query(Instance).filter(
+                            Instance.id == link_props[prop_name]).one()

```

Comment:
Can you please explain what are you trying to do here?

Reply:
Finds the instance which is represented by the client provided link(`link_props[prop_name]` contains the id of the instance after preprocessing.) If there is no such valid instance then we can raise an appropriate error and if there is such an instance then we can go ahead and add a new triple to store this relation.